osh/cmd_eval.py

```diff
@@ -190,6 +190,67 @@ def _HasManyStatuses(node):
     return node
 
 
+def ListInitializeTarget(old_val,
+                         initializer,
+                         has_plus,
+                         blame_loc,
+                         destructive=True):
+    # type: (value_t, value.InitializerList, bool, loc_t, bool) -> value_t
+    UP_old_val = old_val
+    with tagswitch(old_val) as case:
+        if case(value_e.Undef):
+            return bash_impl.BashArray_New()
+        elif case(value_e.Str):
+            if has_plus:
+                e_die("Can't append array to string")
+
+            return bash_impl.BashArray_New()
+        elif case(value_e.BashArray):
+            old_val = cast(value.BashArray, UP_old_val)
+            if not destructive:
+                if has_plus:
+                    old_val = bash_impl.BashArray_Copy(old_val)
+                else:
+                    old_val = bash_impl.BashArray_New()
+            return old_val
+        elif case(value_e.BashAssoc):
+            # OSH compatibility: assoc=(1 2) will create a new array.  This
+            # code will soon be removed when the initialization of the form
+            # "assoc=(key value)" is supported.
+            if not has_plus and len(
```

Comment:
ditto, this could be wrapped better with `()` around the entire condition

Reply:
28bf152247